packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1634,16 +1637,29 @@ Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and
   ///
   /// If no flag named [name] was added to the [ArgParser], an [ArgumentError]
   /// will be thrown.
-  bool boolArg(String name) => argResults![name] as bool;
+  bool boolArg(String name, {bool global = false}) {
+    if (global) {
```

Comment:
Do we need to explicitly pass if we're checking for a global arg, or would it be safe to first check `if (globalResults![name] == null)` and if so, fall back to `argResults`?

Reply:
Okay so took me forever to figure out, but we can't do it like this because when you access a key via [], if it doesn't find it, it will throw an error.
https://github.com/dart-lang/args/blob/a9543c021f9409832b1668f9256f247585362389/lib/src/arg_results.dart#L64-L66

We could alternatively, loop through the options to check for it, though.

```
if (globalResults!.options.contains(name)) {
  return globalResults![name] as bool;
}
```

Reply:
ahh yeah, forgot about that.our snippet looks good to me.

Reply:
Oof. Okay so that snippet doesn't work either because options get filtered out from the visible `options` attribute if they default to null. And if it tries to access a global option key from the `argResults` (and not the `globalResults`), it will throw the previously linked error because it can't find the key.
https://github.com/dart-lang/args/blob/a9543c021f9409832b1668f9256f247585362389/lib/src/arg_results.dart#L85